src/lib/button/button.ts

```diff
@@ -141,13 +141,17 @@ export class MdButton implements OnDestroy {
 
   _setElementColor(color: string, isAdd: boolean) {
     if (color != null && color != '') {
-      this._renderer.setElementClass(this._getHostElement(), `mat-${color}`, isAdd);
+      if (isAdd) {
```

Comment:
I think you can kill this entire function now and just do the null check in `_updateColor` and call `addClass` / `removeClass` directly there.